contracts/StakeManagerV2.sol

```diff
@@ -0,0 +1,351 @@
+
+        address preferredOperator = OPERATOR_REGISTRY
+            .preferredWithdrawalOperator();
+        uint256 operatorsLength = OPERATOR_REGISTRY.getOperatorsLength();
+        if (operatorsLength == 0) revert NoOperatorsAvailable();
+
+        uint256 currentIdx = findOperatorIndex(
+            preferredOperator,
+            operatorsLength
+        );
+
+        while (leftAmount2WithdrawInBnb > 0) {
+            leftAmount2WithdrawInBnb -= _undelegate(
+                OPERATOR_REGISTRY.getOperatorAt(currentIdx),
+                leftAmount2WithdrawInBnb, 
+                uuid
+            );
+
+            currentIdx = currentIdx + 1 < operatorsLength ? currentIdx + 1 : 0;
+        }
+
+        emit RequestedWithdrawal(totalBnbXToBurn, totalAmount2WithdrawInBnb);
+
+        totalBnbXToBurn = 0;
+    }
+
+    /// @notice Claims the withdrawn BNB after the unbonding period.
+    /// @param _idx The index of the withdrawal request.
+    /// @return The amount of BNB claimed.
+    function claimWithdrawal(
+        uint256 _idx
+    ) external override whenNotPaused nonReentrant returns (uint256) {
+        WithdrawalRequest[] storage userRequests = userWithdrawalRequests[
+            msg.sender
+        ];
+        if (userRequests.length == 0) revert NoWithdrawalRequests();
+        if (_idx >= userRequests.length) revert InvalidIndex();
+
+        uint256 totalBnbToWithdraw_ = botUndelegateRequest.amount;
+        uint256 totalBnbXToBurn_ = botUndelegateRequest.amountInBnbX;
+        uint256 amount = (totalBnbToWithdraw_ * amountInBnbX) /
+            totalBnbXToBurn_;
+
+        WithdrawalRequest memory userRequest = userRequests[_idx];
+        uint256 amountToClaim = userRequest.bnbAmount;
+        if (block.timestamp < userRequest.unlockTime) revert Unbonding();
+
+        STAKE_HUB.claim(userRequest.operator, 0);
+
+        // Swap with the last item and pop it.
+        userRequests[_idx] = userRequests[userRequests.length - 1];
+        userRequests.pop();
+
+        (bool success, ) = payable(msg.sender).call{value: amountToClaim}("");
+        if (!success) revert TransferFailed();
+
+        emit ClaimedWithdrawal(msg.sender, _idx, amountToClaim);
+        return amountToClaim;
+    }
+
+    /// @notice Redelegate staked BNB from one operator to another.
+    /// @param _fromOperator The address of the operator to redelegate from.
+    /// @param _toOperator The address of the operator to redelegate to.
+    /// @param _amount The amount of BNB to redelegate.
+    function redelegate(
+        address _fromOperator,
+        address _toOperator,
+        uint256 _amount
+    ) external override nonReentrant onlyRole(MANAGER_ROLE) {
+        if (_fromOperator == address(0)) revert ZeroAddress();
+        if (_toOperator == address(0)) revert ZeroAddress();
+        if (_fromOperator == _toOperator) revert InvalidIndex();
+        if (!OPERATOR_REGISTRY.operatorExists(_fromOperator))
```

Comment:
not required, as if _fromOperator wouldn't exist then delegation amount will be 0 and stakeHub will revert

-----------

but for an extra safety, we can let it be here. wdyt ?